src/help/help_topic_generators.cpp

```diff
@@ -646,149 +670,171 @@ std::string unit_topic_generator::operator()() const {
 			}
 
 			if (info.union_type().size() == 1 && info.union_type()[0] == info.number() && info.is_nonnull()) {
-				std::vector<item> row;
-				const std::string& name = info.name();
-				const std::string& id = info.id();
-				const int views = movement_type.vision_cost(terrain);
-				const int jams  = movement_type.jamming_cost(terrain);
-
-				bool high_res = false;
-				const std::string tc_base = high_res ? "images/buttons/icon-base-32.png" : "images/buttons/icon-base-16.png";
-				const std::string terrain_image = "icons/terrain/terrain_type_" + id + (high_res ? "_30.png" : ".png");
-
-				const std::string final_image = tc_base + "~RC(magenta>" + id + ")~BLIT(" + terrain_image + ")";
-
-				row.emplace_back("<img>src='" + final_image + "'</img> " +
-						make_link(name, "..terrain_" + id),
-					font::line_width(name, normal_font_size) + (high_res ? 32 : 16) );
-
-				//defense  -  range: +10 % .. +70 %
-				const int defense = 100 - movement_type.defense_modifier(terrain);
-				std::string color;
-				if (defense <= 10) {
-					color = "red";
-				} else if (defense <= 30) {
-					color = "yellow";
-				} else if (defense <= 50) {
-					color = "white";
+				terrain_movement_info movement_info = 
+				{
+					info.name(), // 0 name (t_string)
+					info.id(),  // 1 id (t_string)
+					100 - movement_type.defense_modifier(terrain), // 2 defense (int)
+					moves, // 3 movement_cost (int)
+					movement_type.vision_cost(terrain), // 4 vision_cost (int)
+					movement_type.jamming_cost(terrain), // 5 jamming_cost (int)
+					movement_type.get_defense().capped(terrain) // 6 defense_cap (bool)
+				};
+
+				terrain_moves.insert(movement_info);
+			}
+		}
+
+		for(const terrain_movement_info &terr_move : terrain_moves)
+		{
+			std::vector<item> row;
+			const std::string& name = terr_move.name; // get name
+			const std::string& id = terr_move.id; // get id
+			const int defense = terr_move.defense; // get defense
+			const int moves = terr_move.movement_cost; // get moves
+			const int views = terr_move.vision_cost; // get vision cost
+			const int jams  = terr_move.jamming_cost; // get jamming cost
+			const bool has_cap = terr_move.defense_cap; // has defense cap
+			const bool cannot_move = moves > type_.movement();
+
+
+			bool high_res = false;
+			const std::string tc_base = high_res ? "images/buttons/icon-base-32.png" : "images/buttons/icon-base-16.png";
+			const std::string terrain_image = "icons/terrain/terrain_type_" + id + (high_res ? "_30.png" : ".png");
+
+			const std::string final_image = tc_base + "~RC(magenta>" + id + ")~BLIT(" + terrain_image + ")";
+
+			row.emplace_back("<img>src='" + final_image + "'</img> " +
+					make_link(name, "..terrain_" + id),
+				font::line_width(name, normal_font_size) + (high_res ? 32 : 16) );
+
+			//defense  -  range: +10 % .. +70 %
+			//const int defense = 100 - movement_type.defense_modifier(terrain); // defense
```

Comment:
Why is this commented out? If it's not needed, remove it completely. If it _is_ needed, why is it commented?

Reply:
My bad on the general lack of cleanup. I'll go through and tear out the unused bits / comments and squash tonight.